src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -169,16 +158,14 @@ protected void SetParentOperation(IOperation? parent)
 
             foreach (var operation in operations)
             {
-                // Due to lazy construction of nodes, it's very possible that two threads might see lists where
-                // the first operation in an array is reference equal between the arrays, but the second operation
-                // is not reference equal. This difference is generally not observable from a user perspective (unless
-                // they're doing their own reference equality), and fixing it by removing laziness from IOperation
-                // entirely is tracked by https://github.com/dotnet/roslyn/issues/35818. Until then, however, we've
-                // seen real world examples where bailing out of the entire set operation causes infinite loops when
-                // searching for a parent operation, so we only skip setting the parent for a single node at a time.
                 if ((operation as Operation)?._parentDoNotAccessDirectly != s_unset)
                 {
+#if DEBUG
+                    VerifyParentOperation(parent, operation);
                     continue;
+#else
+                    break;
```

Comment:
>break; [](start = 20, length = 6)

Why is it safe to skip remaining operations?

Reply:
Because this is entirely non-lazy now, there's no chance that we can observe an array where some of the elements were constructed before, and some later. They're all fresh nodes, so if one is set, they were all set. I actually think I can simplify the code more: nodes seen by `SetParentOperation` should never have existing parents, and this `if` is entirely unnecessary.

---
In reply to: [526911026](https://github.com/dotnet/roslyn/pull/49453#discussion_r526911026) [](ancestors = 526911026)